pkg/catalogv2/git/git.go

```diff
@@ -125,9 +126,22 @@ func (r *Repository) setRepoCredentials() error {
 			return fmt.Errorf("failed to parse ssh private key: %w", err)
 		}
 
+		parts := strings.Split(r.URL, "@")
```

Comment:
When happens when the user doesn't provide a secret ?

Reply:
Added unit-tests and improved the isGitSSH method with another regex pattern